### PR TITLE
adaptive moments and pgauss smoothing

### PR DESCRIPTION
**Pull request by esheldon**

### CODE REVIEW

File: metadetect/fitting.py [hunk 1]
```diff
@@ -213,6 +216,12 @@ def _fit_obs(
         res["obj_res"] = fitter.go(obs)
         res["psf_res"] = fitter.go(obs.psf, **psf_go_kwargs)
 
+        if fitter.kind == 'am':
+            res["obj_res"]["mom"] = res["obj_res"]["sums"]
+            res["obj_res"]["mom_cov"] = res["obj_res"]["sums_cov"]
+            res["psf_res"]["mom"] = res["psf_res"]["sums"]
+            res["psf_res"]["mom_cov"] = res["psf_res"]["sums_cov"]
+
```
Comment (beckermr) [inline metadetect/fitting.py:224]: Are the normalizations here the same? We may want to upstream this change into ngmix itself.

**Comment by esheldon** [comment]
https://www.githubstatus.com/incidents/qzmhbwp7f6nn

File: metadetect/fitting.py (hunk 1 above)
Comment (esheldon) [reply metadetect/fitting.py:224]: the AM sums are not normalized by flux

Comment (esheldon) [reply metadetect/fitting.py:224]: Is that the same as prepsfmom?

Comment (beckermr) [reply metadetect/fitting.py:224]: Yes that should be the same. I will double check.

Comment (beckermr) [reply metadetect/fitting.py:224]: These norms are the same. We should move this bit of code into ngmix.

**Comment by esheldon** [comment]
shall we merge this after the other PR to get the ngmix upstream fixes?

File: metadetect/fitting.py [hunk 2]
```diff
@@ -225,12 +225,6 @@ def _fit_obs(
         res["obj_res"] = fitter.go(obs)
         res["psf_res"] = fitter.go(obs.psf, **psf_go_kwargs)
 
-        if fitter.kind == 'am':
-            res["obj_res"]["mom"] = res["obj_res"]["sums"]
-            res["obj_res"]["mom_cov"] = res["obj_res"]["sums_cov"]
-            res["psf_res"]["mom"] = res["psf_res"]["sums"]
-            res["psf_res"]["mom_cov"] = res["psf_res"]["sums_cov"]
-
```
Comment (beckermr) [inline metadetect/fitting.py]: We'll want to add sums_norm in ngmix before releasing 2.1

**Comment by beckermr** [comment]
You likely want to merge master into this branch to get the tests to pass with old ngmix versions. 

Also for AM, you'll need to detect the ngmix version and use the old `mom` name if you want this to run with older ngmix. You can import MOMNAME from fitting and that has the right prefix.

**Comment by esheldon** [comment]
I did merge master

**Comment by beckermr** [comment]
Ack I missed the master merge. Admom is only one band so we don't care about sums_norm in principle.

File: metadetect/fitting.py (hunk 2 above)
Comment (beckermr) [reply metadetect/fitting.py]: We can skip this if we'd like actually.

**Comment by beckermr** [comment]
So the test failure is from passing fwhm_smooth to older versions of the fitters that did not support that. You can avoid this by passing **kwargs if the smoothing is > 0 or you can simply wait for ngmix 2.1

**Comment by beckermr** [comment]
Shall we merge?

**Pull request merged**